app/src/main/java/org/systers/mentorship/view/activities/SendRequestActivity.kt

```diff
@@ -106,8 +109,69 @@ class SendRequestActivity: BaseActivity() {
                         notes = notes,
                         endDate = endDate
                 )
+                /*
+                 make API call, add observer and see if similar field exists.
+                 conditions for failure: receiver, mentor/mentee field, end date are same
+                 */
+                lateinit var requestsList: List<Relationship>
+                val requestsViewModel = ViewModelProviders.of(this)
+                        .get(RequestsViewModel::class.java)
+                requestsViewModel.successful.observe(this, Observer {
+                    successful ->
+                    if (successful != null) {
+                            if (successful) {
+                                requestsList = requestsViewModel.allRequestsList
+                                requestsList.filter {
+                                    val isPendingState = RelationState.PENDING.value == it.state
+                                    val hasEndTimePassed = getUnixTimestampInMilliseconds(it.endsOn) < System.currentTimeMillis()
+                                    isPendingState && !hasEndTimePassed
+                                }
+                                //find if similar relation exists
+                                for (relationship in requestsList){
```

Comment:
This is a rather expensive way of doing this. Instead, you could just check if the mentor/mentee is available for a new mentor-mentee relationship. 

But this will also work.

Reply:
Done please review : )